lib/chopperbot/message_builder.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Chopperbot.MessageBuilder do
+  alias Chopperbot.Split
+
+  @callback validate_text_input(String.t()) :: {:ok, String.t()} | {:error, String.t()}
+  @callback build_ok_message([Split.Order.t()]) :: map()
+  @callback build_error_message(String.t()) :: map()
+
+  @doc """
+  Build a message from the given text and platform.
+  """
+  @spec build(String.t(), for: atom()) :: map()
+  def build(text, for: platform) do
+    message_builder = get_message_builder_from_platform(platform)
+
+    with {:ok, validated_text} <- message_builder.validate_text_input(text),
```

Comment:
Not sure if it is common for `validate` to alter the input? Should we call `parse` instead?

Reply:
Sounds better 👍 
How about `normalize` lol

Reply:
Yeah, sounds good na